lib/cloud_controller/diego/docker/lifecycle_protocol.rb

```diff
@@ -22,10 +22,18 @@ def staging_action_builder(config, staging_details)
           end
 
           def task_action_builder(config, task)
+            if task.run_action_user == 'root' && !Config.config.get(:allow_docker_root_user)
+              raise ::CloudController::Errors::ApiError.new_from_details('RunnerError', "'root' user not permitted.")
```

Comment:
This error message may be confusing/unclear to users, since they may not know their app is trying to run as root.

It'd be more luxurious to have something _like_ `attempting to run task as 'root' user, which is not permitted.`